### PR TITLE
[MASWE-0009] Add Weak Cryptographic Key Generation (by appknox)

### LINKED ISSUE #2573: New Risk - Weak Cryptographic Key Generation [weak-crypto-key-generation]
## Description

Create a new risk for "Weak Cryptographic Key Generation (MASVS-CRYPTO-2)" using the following information:

e.g. 1024-bit RSA keys, 128-bit AES keys, 160-bit ECDSA keys, 80-bit symmetric keys

Create "`risks/MASVS-CRYPTO/2-***-****/weak-crypto-key-generation/risk.md`" including the following content:

```yaml
---
title: Weak Cryptographic Key Generation
alias: weak-crypto-key-generation
platform: [android, ios]
profiles: [L1, L2]
mappings:
  masvs-v1: [MSTG-CRYPTO-2]
  masvs-v2: [MASVS-CRYPTO-2]
  mastg-v1: [MASTG-TEST-0061, MASTG-TEST-0014]

---

## Overview

## Impact

## Modes of Introduction

## Mitigations

```

**To complete the sections follow the guidelines from [Writing MASTG Risks & Tests](https://docs.google.com/document/d/1EMsVdfrDBAu0gmjWAUEs60q-fWaOmDB5oecY9d9pOlg/edit?usp=sharing)**

Use at least the following references:

- <https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-57pt1r5.pdf>
- <https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-131Ar2.pdf>

When creating the corresponding tests, use the following areas to guide you:

- insufficient Key Length

**MASTG v1 Refactoring:**

If the risk has a MASVS v1 ID, you can use it to [search for related tests in the MASTG](https://mas.owasp.org/MASTG/tests/) and use them as input to define your risks and associated tests.

- [MASTG-TEST-0061 - Verifying the Configuration of Cryptographic Standard Algorithms (ios)](https://mas.owasp.org/MASTG/tests/android/MASVS-CRYPTO/MASTG-TEST-0061/)
- [MASTG-TEST-0014 - Testing the Configuration of Cryptographic Standard Algorithms (android)](https://mas.owasp.org/MASTG/tests/android/MASVS-CRYPTO/MASTG-TEST-0014/)

## Acceptance Criteria

- [ ] The risk has been created in the correct directory (`risks/MASVS-CRYPTO/2-***-****/weak-crypto-key-generation/risk.md`)
- [ ] The risk content follows the [guidelines](https://docs.google.com/document/d/1EMsVdfrDBAu0gmjWAUEs60q-fWaOmDB5oecY9d9pOlg/edit?usp=sharing)
- [ ] At least one GitHub Issue has been created for the corresponding tests (derived from "Modes of Introduction")
- [ ] The risk indicates the related MASTG v1 tests in its metadata.

### PR DESCRIPTION
**Pull request by sk3l10x1ng**
closes #2573

### CODE REVIEW

File: risks/MASVS-CRYPTO/2-weak-crypto-key-generation/weak-key-generation/android-weak-key-generation/rules/mastg-weak-crypto-key-generation.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
+rules:
```
Comment (sushi2k) [inline risks/MASVS-CRYPTO/2-weak-crypto-key-generation/weak-key-generation/android-weak-key-generation/rules/mastg-weak-crypto-key-generation.yml:1]: Let's add the source, if it wasn't done by us from scratch.

```suggestion
# original source: https://github.com/MobSF/mobsfscan/blob/main/mobsfscan/rules/semgrep/crypto/weak_key_size.yaml
rules:
```

File: risks/MASVS-CRYPTO/2-weak-crypto-key-generation/weak-key-generation/android-weak-key-generation/demo-1/output.sarif [hunk 2]
```diff
@@ -0,0 +1,135 @@
+{
```
Comment (sushi2k) [inline risks/MASVS-CRYPTO/2-weak-crypto-key-generation/weak-key-generation/android-weak-key-generation/demo-1/output.sarif:1]: Can you please remove the SARIF file for now? We can add later to keep the PR cleaner.

**Review by sushi2k — CHANGES_REQUESTED** [review]
Thanks you @sk3l10x1ng for the PR! Have a look at the comments. The main change that is needed is to not mention AES-128 as insecure algorithm. You could use AES-64 instead as example. Let me know if any questions

**Comment by sk3l10x1ng** [comment]
> Thanks you @sk3l10x1ng for the PR! Have a look at the comments. The main change that is needed is to not mention AES-128 as insecure algorithm. You could use AES-64 instead as example. Let me know if any questions

@sushi2k 
- We have created the test case on AES-128 bit based on the following examples  provided in the issue #2573 . 
- Can we use DES algorithm for the test case for the ios . Where AES-64 is not possible because the contain key sizes 128, 192, 256

**Comment by sushi2k** [comment]
Hi @sk3l10x1ng. You could use 3DES, as this is algorithm is deprecated, instead of AES-128. You are right AES-128 is listed as example in the issue, but this is not correct.
Can you please also go through my comments and follow-up on them? Regarding the test cases and weaknesses we have a new structure which can be found here: <https://docs.google.com/document/d/1EMsVdfrDBAu0gmjWAUEs60q-fWaOmDB5oecY9d9pOlg/edit?usp=sharing>. Can you please update the content to this new format?  Let me know if any questions. 

Thanks for your support.

**Comment by sk3l10x1ng** [comment]
@cpholguera @sushi2k . updated according to new file structure in #2849

**Pull request closed without merging**